builtin/game/item.lua

```diff
@@ -530,6 +530,18 @@ function core.node_dig(pos, node, digger)
 	end
 end
 
+function core.itemstring_with_palette(item, palette_index)
+	local stack = ItemStack(item) -- convert to ItemStack
```

Comment:
This method should be used when some algorithm accepts metadata, but only accepts itemstrings. Craft recipe registration is the only place I know of where this is true.

In most cases the engine accepts itemstacks, and then the metadata can be directly accessed without these methods.